### PR TITLE
Add Dataflow TensorRT Bert Q&A Sample

### PR DESCRIPTION
**Pull request by mengdong**
This commit adds GCP Dataflow TensorRT sample, with an example model of BERT large. @roarjn, could you please help review if any modification needed?

### CODE REVIEW

File: dataflow-samples/bert-qa-trt-dataflow/README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  --worker_harness_container_image "gcr.io/${PROJECT_ID}/trt-dataflow:latest"          \
+  --experiment "worker_accelerator=type:nvidia-tesla-t4;count:1;install-nvidia-driver" \
+  --experiment "use_runner_v2"                            \
+  --subnetwork "regions/${REGION}/subnetworks/default"    \
+  --autoscaling_algorithm=NONE                            \
+  --num_workers=1                                         \
+  --number_of_worker_harness_threads=4  
+```
+
+We expect ~100 GPS(Query per second) on a T4 n1-standard-4 instance. Which calculates to 650k inference per dollar on a 345 million parameters model. This could be further optimized with analysis of beam performance and TensorRT API usage. To understand the sample, we recommend go through following materials:
```
Comment (roarjn) [inline dataflow-samples/bert-qa-trt-dataflow/README.md:30]: ~100 QPS

File: dataflow-samples/bert-qa-trt-dataflow/README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
+# TensorRT Bert Q&A Inference in GCP Dataflow
+
+## Sample Overview
+
+This is sample that deployed a optimized BERT large model with sequence-length 384, fine-tuned with squad Q&A task, then run inference with GCP dataflow. To run the sample, please first create a docker image and push to GCR registry with [build_and_push.sh](build_and_push.sh). Please make sure your git lfs is enabled so that you can download the engine file. In the sample code, we generate replicated documentation and questions to send, each request has a default batch size of 16 which match the optimial batch size of the TensorRT Engine profile. 
+
```
Comment (roarjn) [inline dataflow-samples/bert-qa-trt-dataflow/README.md:6]: `sh build_and_push.sh`

File: dataflow-samples/bert-qa-trt-dataflow/README.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+Once image is pushed successfully, make sure your local python environment has version 3.6 and properly installed with dataflow prerequsite, then dataflow job could be launch with:
+```
+export PROJECT_ID=[your gcp project ID]
+export GCS_BUCKET=[your gcs bucket for temp location]
+export REGION=us-central1
```
Comment (roarjn) [inline dataflow-samples/bert-qa-trt-dataflow/README.md:11]: # GPUs by region https://cloud.google.com/compute/docs/gpus/gpu-regions-zones

File: dataflow-samples/bert-qa-trt-dataflow/README.md [hunk 4]
```diff
@@ -0,0 +1,37 @@
+# TensorRT Bert Q&A Inference in GCP Dataflow
+
+## Sample Overview
+
+This is sample that deployed a optimized BERT large model with sequence-length 384, fine-tuned with squad Q&A task, then run inference with GCP dataflow. To run the sample, please first create a docker image and push to GCR registry with [build_and_push.sh](build_and_push.sh). Please make sure your git lfs is enabled so that you can download the engine file. In the sample code, we generate replicated documentation and questions to send, each request has a default batch size of 16 which match the optimial batch size of the TensorRT Engine profile. 
+
+Once image is pushed successfully, make sure your local python environment has version 3.6 and properly installed with dataflow prerequsite, then dataflow job could be launch with:
```
Comment (roarjn) [inline dataflow-samples/bert-qa-trt-dataflow/README.md:7]: Add steps needed for user to launch from local environment
`pip install apache-beam`

File: dataflow-samples/bert-qa-trt-dataflow/README.md [hunk 5]
```diff
@@ -0,0 +1,37 @@
+# TensorRT Bert Q&A Inference in GCP Dataflow
+
+## Sample Overview
+
+This is sample that deployed a optimized BERT large model with sequence-length 384, fine-tuned with squad Q&A task, then run inference with GCP dataflow. To run the sample, please first create a docker image and push to GCR registry with [build_and_push.sh](build_and_push.sh). Please make sure your git lfs is enabled so that you can download the engine file. In the sample code, we generate replicated documentation and questions to send, each request has a default batch size of 16 which match the optimial batch size of the TensorRT Engine profile.
```
Comment (roarjn) [inline dataflow-samples/bert-qa-trt-dataflow/README.md:5]: `sh build_and_push.sh my_project_id`

**Review by roarjn — COMMENTED** [review]
added some minor clarification comments

File: dataflow-samples/bert-qa-trt-dataflow/README.md (hunk 4 above)
Comment (mengdong) [reply dataflow-samples/bert-qa-trt-dataflow/README.md:9]: added getting started guide

File: dataflow-samples/bert-qa-trt-dataflow/README.md (hunk 2 above)
Comment (mengdong) [reply dataflow-samples/bert-qa-trt-dataflow/README.md:8]: fixed

File: dataflow-samples/bert-qa-trt-dataflow/README.md (hunk 3 above)
Comment (mengdong) [reply dataflow-samples/bert-qa-trt-dataflow/README.md:13]: added

File: dataflow-samples/bert-qa-trt-dataflow/README.md (hunk 5 above)
Comment (mengdong) [reply dataflow-samples/bert-qa-trt-dataflow/README.md]: fixed

**Comment by mengdong** [comment]
@roarjn Thanks for the review, address all issues per comment, any other items?

**Comment by arunraman** [comment]
@mengdong is it fine to upload the model file to Github or can we make it download from NGC?

**Comment by mengdong** [comment]
NGC couldn't host a public model at will. I don't know there are rules against putting the model in github? I think this is the easiest way for anyone to jump on the sample. Model is trained with all public data.

**Review by arunraman — APPROVED** [review]
ok, Then its fine.

**Pull request merged**